pkg/db/db_session/default.go

```diff
@@ -112,20 +151,33 @@ func waitForNotification(ctx context.Context, l *pq.Listener, callback func(id s
 		case <-time.After(10 * time.Second):
 			logger.V(10).Infof("Received no events on channel during interval. Pinging source")
 			go func() {
+				// TODO: Need to handle the error, especially in cases of network failure.
 				l.Ping()
```

Comment:
See https://github.com/openshift-online/maestro/pull/231/files#r1887602957 

We'll need to ensure we refresh the token to limit issues related to the token being expired

Reply:
Yes. that is why I added a TODO for the connection failure. Will have a separated PR for this one.

Reply:
It has to be a fast follow because if the error occurs it'll be hard to debug.
Perhaps we could start by logging the error in this PR